lib/s3.py

```diff
@@ -40,6 +50,37 @@ def is_key_present(url):
     return get_key(url.hostname) is not None
 
 
+def sign_request(url, method='GET', checksum=SHA256_NIL, headers={}, region='any', service='S3'):
+    access_key, secret_key = get_key(url.hostname)
```

Comment:
This would `ValueError` if get_key returns None, i.e. there is no key. I suppose that's ok, but the docstring should maybe point out that this needs to be asserted before? In the spirit of robustness, perhaps raise a more useful error than "not enough values to unpack"?

Reply:
The plan is that this would only ever be called after we checked for a key.  I'll document that precondition, rather than change the code here.

On the other hand, this will indeed fail very ungracefully for an invalidly-formatted key file.  I'll fix that.